File: electronic_drums.cs
// electronic_drums.cs — C# версия

using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Threading;

class ElectronicDrums
{
    static Dictionary<string, (string name, int freq)> drums = new Dictionary<string, (string, int)>
    {
        {"1", ("Бас", 80)}, {"2", ("Снэр", 200)}, {"3", ("Хай-хэт", 1000)},
        {"4", ("Том 1", 150)}, {"5", ("Том 2", 120)}, {"6", ("Райд", 800)},
        {"7", ("Крэш", 600)}, {"8", ("Перкуссия", 400)}
    };

    static void PlaySound(int freq, int duration)
    {
        try
        {
            Process.Start("beep", $"-f {freq} -l {duration}");
        }
        catch
        {
            Console.Beep(freq, duration);
        }
    }

    static void Main()
    {
        Console.WriteLine("\u001B[36m🥁 Electronic Drum Kit (C#)\u001B[0m");
        Console.WriteLine("Клавиши для игры:");
        Console.WriteLine("  [1] Бас   [2] Снэр   [3] Хай-хэт");
        Console.WriteLine("  [4] Том1  [5] Том2   [6] Райд");
        Console.WriteLine("  [7] Крэш  [8] Перк");
        Console.WriteLine("Нажмите q для выхода");

        while (true)
        {
            Console.Write("\u001B[33m> \u001B[0m");
            string input = Console.ReadLine().Trim().ToLower();
            if (input == "q") break;
            if (drums.ContainsKey(input))
            {
                var d = drums[input];
                Console.WriteLine($"Играем {d.name}");
                for (int i = 0; i < 3; i++)
                {
                    PlaySound(d.freq + (int)(d.freq * 0.02 * i), 30);
                    Thread.Sleep(10);
                }
            }
            else
            {
                Console.WriteLine("Неизвестная команда.");
            }
        }
    }
}

File: electronic_drums.rs
// electronic_drums.rs — Rust версия

use std::collections::HashMap;
use std::io::{self, Write};
use std::process::Command;
use std::thread;
use std::time::Duration;
use colored::*;

struct Drum {
    name: String,
    freq: u32,
}

fn main() {
    let mut drums = HashMap::new();
    drums.insert("1", Drum { name: "Бас".to_string(), freq: 80 });
    drums.insert("2", Drum { name: "Снэр".to_string(), freq: 200 });
    drums.insert("3", Drum { name: "Хай-хэт".to_string(), freq: 1000 });
    drums.insert("4", Drum { name: "Том 1".to_string(), freq: 150 });
    drums.insert("5", Drum { name: "Том 2".to_string(), freq: 120 });
    drums.insert("6", Drum { name: "Райд".to_string(), freq: 800 });
    drums.insert("7", Drum { name: "Крэш".to_string(), freq: 600 });
    drums.insert("8", Drum { name: "Перкуссия".to_string(), freq: 400 });

    println!("{}", "🥁 Electronic Drum Kit (Rust)".cyan());
    println!("Клавиши для игры:");
    println!("  [1] Бас   [2] Снэр   [3] Хай-хэт");
    println!("  [4] Том1  [5] Том2   [6] Райд");
    println!("  [7] Крэш  [8] Перк");
    println!("Нажмите q для выхода");

    loop {
        print!("\x1b[33m> \x1b[0m");
        io::stdout().flush().unwrap();
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let input = input.trim().to_lowercase();
        if input == "q" { break; }

        if let Some(drum) = drums.get(input.as_str()) {
            println!("Играем {}", drum.name);
            // Имитация реверберации
            for i in 0..3 {
                let _ = Command::new("beep")
                    .args(&["-f", &(drum.freq + (drum.freq as f32 * 0.02 * i as f32) as u32).to_string(), "-l", "30"])
                    .status();
                thread::sleep(Duration::from_millis(10));
            }
        } else {
            println!("Неизвестная команда.");
        }
    }
}
